runtime/interp.c

```diff
@@ -210,11 +210,18 @@ Caml_inline void check_trap_barrier_for_effect
 #define SP_REG asm("%r17")
 #define ACCU_REG asm("%r16")
 #endif
-#ifdef __mc68000__
+#ifdef __mc68000__ /* { */
+#ifdef __linux__
+/* glibc relies upon %a5 to always contain the base address of the GOT. */
+#define PC_REG asm("a3")
+#define SP_REG asm("a4")
+#define ACCU_REG asm("d7")
+#else
 #define PC_REG asm("a5")
 #define SP_REG asm("a4")
 #define ACCU_REG asm("d7")
 #endif
+#endif /* } __mc68000 */
```

Comment:
No need to add more ifdefs, the revised register assignment will work everywhere.
```suggestion
#ifdef __mc68000__
#define PC_REG asm("a3")
#define SP_REG asm("a4")
#define ACCU_REG asm("d7")
#endif
```

Reply:
Ok then.